waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -977,6 +978,7 @@
          (fn? start-scheduler-syncer-fn)
          (or (nil? watch-retries) (integer? watch-retries))]}
   (let [authorizer (utils/create-component authorizer)
+        custom-options (into {} (filter (comp namespace key) context))
```

Comment:
Why are any namespaced keys automatically promoted to custom options? I would prefer we reserve a special key for custom options that maps to a dictionary of values, e.g.
```:custom-optiiosn {}```

Reply:
That does seem more straightforward. I'll refactor to do that instead.